lib/src/main/ktx/com/fabernovel/statefullayout/Lifecycle.kt

```diff
@@ -0,0 +1,30 @@
+package com.fabernovel.statefullayout
+
+import android.view.View
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+
+internal val View.viewCoroutineScope: CoroutineScope
+    get() {
```

Comment:
To avoid concurrency issue, could we throw an error if this getter is not call from the UI thread (usually we manipulate view concepts from UI thread)